services/sources/publisher_backlist_service.py

```diff
@@ -0,0 +1,65 @@
+from datetime import datetime, timedelta, timezone
+import os
+import requests
+import json
+from typing import Optional
+
+from logger import create_log
+from mappings.UofM import UofMMapping
+from .source_service import SourceService
+
+
+logger = create_log(__name__)
+
+
+class PublisherBacklistService(SourceService):
+    def __init__(self):
+        self.airtable_auth_token = os.environ.get('AIRTABLE_KEY', None)
+
+        pass
+
+    def get_records(
+        self,
+        full_import: bool=False, # get all records from Airtable
+        start_timestamp: datetime=None,
+        offset: Optional[int]=None, # start at specific position in paginated results 
+        limit: Optional[int]=None
+    ) -> list[UofMMapping]:
+        
+        json_data = self.get_publisher_backlist_records(full_import, start_timestamp, offset, limit)
+
+        array_index = 0
+        while array_index != len(json_data):
+            for dict_index in range(0, len(json_data['records'])):
+                metaDict = json_data['records'][dict_index]
```

Comment:
I recommend actually putting this down in the other method so that this function only gets a flat list of json records and then can map those records.

Reply:
I'm kinda confused here since if I move this code down to the other method wouldn't I have to return the `metaDict `to `get_records()` and therefore the conditional would end after only returning the first record json file in the array?